### DISCUSSION TITLE
connect to multiple instances of Jellyfin at once?

### DISCUSSION THREAD

**Discussion by mmattbtw** [rekkyrosso/ampcast#18] [Ideas]
perhaps an odd use case, but both me and my friend have Jellyfin servers that we share out to each other. we both have users on each other's servers and it'd be cool to have access to both servers at the same time in ampcast!

(also, such a sick application keep up the work)

**Comment by rekkyrosso** [comment]
I'll have a think about that.

My initial thoughts are that it would probably be a lot of work for a niche use case. But I'm not against the idea.

**Reply by mmattbtw** [reply]
i might try my hand at it in a fork, just to get like a proof of concept going.

**Reply by rekkyrosso** [reply]
Take a look at how I implemented Subsonic with a factory pattern so that I can generate the same "flavour" of API just with different instances (Airsonic, Ampache and Gonic). It's not exactly the same thing but it's a start.

**Reply by rekkyrosso** [reply]
I think the `src` property of my media objects is the thing that will trip you up the most:

https://github.com/rekkyrosso/ampcast/blob/main/src/types/BaseMediaObject.d.ts

**Comment by facolors** [comment]
seconding this but for navidrome
